.github/workflows/lint.yml

```diff
@@ -0,0 +1,19 @@
+name: Linter
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install modules
+      run: npm i
+    - name: Run ESLint
+      run: node_modules/.bin/next lint --quiet
+    - name: Run Prettier
+      run: node_modules/.bin/prettier --check "**/*.{js,css,md,json,jsx,ts,tsx}"
```

Comment:
Both are in the same job because no point in doing npm install in two separate workflows.